### PR TITLE
Small fixes and additions to the security best practices

### PR DESCRIPTION
**Pull request by gwillem**
## Purpose of this pull request

This pull request fixes a few typos and adds common security best practices.

## Magento release version

Applies to all versions.

## Product editions

Is this update specific to a product edition?

- [x] Magento Open Source only
- [x] Magento Commerce only

Is this update specific to an installed feature extension

No

## Affected documentation pages

- https://docs.magento.com/user-guide/magento/magento-security-best-practices.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:30]:
```suggestion
- Make sure that the server operating system is secure. Work with your hosting provider to ensure that there is no unnecessary software running on the server.
```
```suggestion
- Make sure that the server operating system is secure. Work with your hosting provider to ensure that there is no unnecessary software running on the server.
```
```suggestion
- Make sure that the server operating system is secure. Work with your hosting provider to ensure that no unnecessary software is running on the server.
```

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:33]:
```suggestion
- Use strong and unique passwords with a minimum of eight characters, and change them periodically.  
```

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:41]:
```suggestion
- Limit access to the Magento Admin by updating the allow list with the IP address of each computer that is authorized to use the Admin. For examples of how to add IP addresses to the allow list, see [Secure Your Magento Admin](https://www.sonassi.com/knowledge-base/magento-kb/secure-your-magento-admin/){:target="\_blank"}.
```

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:42]:
```suggestion
- Do not install extensions directly on a production server. 
```

Comment (ccondra) [reply src/magento/magento-security-best-practices.md:42]: The Magento Connect downloader is no longer available.

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [reply src/magento/magento-security-best-practices.md:41]: The Magento Connect downloader is no longer used.

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:52]:
```suggestion
- Do not install third-party web-based database managers on your production server, as they can provide backdoor access to attackers.
```

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:73]:
```suggestion
- Use a [strong](https://support.google.com/accounts/answer/32040?hl=en) password for the Magento Admin.
```

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:95]:
```suggestion
- Run a server-side [Magento malware and vulnerability monitor][12] to quickly alert you of security issues with your files, database, admin accounts, and third party components.
```

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:43]:
```suggestion
- Use two-factor authorization for Admin logins, as required by Magento 2.4.x. For older releases, there are several extensions available that provide additional security and require a one-time passcode that is generated on your phone, or a token from a special device.
```

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:44]:
```suggestion
- Review your server for “development leftovers.” Make sure there are no accessible log files, publicly visible .git directories, tunnels to execute SQL, database dumps, backup archives, `phpinfo` files, or any other unprotected files that are not required and might be used in an attack.
```

**Review by ccondra — COMMENTED** [review]
Hi -- I made a few comments and suggestions that will be reviewed by our security team lead. Thank you!  -- Claire

**Review by ccondra — APPROVED** [review]
Thank you!

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:33]:
```suggestion
- Use [strong](https://support.google.com/accounts/answer/32040?hl=en), unique passwords with a minimum of eight characters, and change them periodically.
```

Comment (ccondra) [reply src/magento/magento-security-best-practices.md:33]: Removed trailing spaces

File: src/magento/magento-security-best-practices.md
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:42]:
```suggestion
- Do not install extensions directly on a production server.
```

Comment (ccondra) [reply src/magento/magento-security-best-practices.md:42]: Removed trailing space.

**Review by ccondra — APPROVED** [review]
Removed a few trailing spaces.

**Comment by ccondra** [comment]
running tests

**Comment by gwillem** [comment]
Thanks for maintaining these docs!

**Pull request merged**
